wapitiCore/attack/mod_csp.py

```diff
@@ -0,0 +1,102 @@
+
+    check_default = ["*", "unsafe-eval"]
+    check_script_src = ["unsafe-inline", "data:", "http", "https"]
+    check_object_src = ["none"]
+    check_base_uri = ["none", "self"]
+
+    all_check = [check_default, check_script_src, check_object_src, check_base_uri]
+
+    def check_policy_values(self, policy_name, check_list, csp_dict):
+        """
+        This function return the status of the tested element in the CSP as an int. Possible values:
+        -1 : the element is missing in the CSP
+        0  : the element is set, but his value is not secure
+        1  : the element is set and his value is secure
+        """
+
+        if policy_name not in csp_dict and "default-src" not in csp_dict:
+            return -1
+
+        # The HTTP CSP "default-src" directive serves as a fallback for the other CSP fetch directives.
+        # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/default-src
+        policy_values = csp_dict.get(policy_name, csp_dict["default-src"])
+
+        # If the tested element is default-src or script-src, we must ensure that none of this unsafe values are present
+        if policy_name in ["default-src", "script-src"]:
+            if any(unsafe_value in policy_values for unsafe_value in check_list):
+                return 0
+        # If the tested element is none of the previous list, we must ensure that one of this safe values is present
+        else:
+            if any(safe_value in policy_values for safe_value in check_list):
+                return 1
+            else:
+                return 0
+
+        return 1
+
+    def attack(self):
+        head = {
+            'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8',
+            'Accept-Encoding': 'gzip, deflate, br'
+        }
+        url = self.persister.get_root_url()
+        request = Request(url)
+        response = self.crawler.get(request, follow_redirects=True, headers=head)
+
+        if "Content-Security-Policy" not in response.headers:
+            self.log_red(Additional.MSG_NO_CSP)
+            self.add_addition(
+                category=Additional.INFO_CSP,
+                level=Additional.LOW_LEVEL,
+                request=request,
+                info=Additional.MSG_NO_CSP
+            )
+        else:
+            csp = response.headers["Content-Security-Policy"]
+            csp_dict = {}
+            regex = re.compile(r"\s*((?:'[^']*')|(?:[^'\s]+))\s*")
+            for policy_string in csp.split(";"):
+                try:
+                    policy_name, policy_values = policy_string.strip().split(" ", 1)
+                except ValueError:
+                    # Either it is malformed or we reach the end
+                    continue
+                csp_dict[policy_name] = [value.strip("'") for value in regex.findall(policy_values)]
+
+            self.log_blue(_("Checking CSP :"))
```

Comment:
In order to comply with other modules behavior I guess we will remove this log and only print useful information. Most modules only print log when something interesting is found.